### PR TITLE
[Superseded] Added carryalls to spectator Economy statistics

### PR DESCRIPTION
**Pull request by penev92**
As suggested on Discord. This would be relevant when spectating D2k matches.

### CODE REVIEW

**Review by abcdefg30 — APPROVED** [review]

File: mods/d2k/chrome/ingame-observer.yaml [hunk 1]
```diff
@@ -770,7 +778,14 @@ Container@OBSERVER_WIDGETS:
 									Align: Right
 									Shadow: True
 								Label@HARVESTERS:
-									X: 555
+									X: 525
```
Comment (PunkPun) [inline mods/d2k/chrome/ingame-observer.yaml:781]:
```suggestion
									X: 555
```

You can't do that! It misalignes the whole UI

File: mods/d2k/chrome/ingame-observer.yaml [hunk 2]
```diff
@@ -770,7 +778,14 @@ Container@OBSERVER_WIDGETS:
...
+									Y: 0
+									Width: 80
+									Height: PARENT_BOTTOM
+									Align: Right
+									Shadow: True
+								Label@CARRYALLS:
+									X: 600
```
Comment (PunkPun) [inline mods/d2k/chrome/ingame-observer.yaml:788]:
```suggestion
									X: 635
```
555 + 80 = 635

File: mods/d2k/chrome/ingame-observer.yaml [hunk 3]
```diff
@@ -384,6 +384,14 @@ Container@OBSERVER_WIDGETS:
 									Text: Harvesters
 									Align: Right
 									Shadow: True
+								Label@CARRYALLS_HEADER:
+									X: 625
```
Comment (PunkPun) [inline mods/d2k/chrome/ingame-observer.yaml:388]:
```suggestion
									X: 635
```
555 + 80 = 635

File: mods/d2k/chrome/ingame-observer.yaml [hunk 4]
```diff
@@ -313,7 +313,7 @@ Container@OBSERVER_WIDGETS:
 						Container@ECONOMY_STATS_HEADERS:
 							X: 0
 							Y: 0
-							Width: 640
+							Width: 740
```
Comment (PunkPun) [inline mods/d2k/chrome/ingame-observer.yaml:316]:
```suggestion
							Width: 720
```

you are expanding UI by 80, not 100

File: mods/d2k/chrome/ingame-observer.yaml [hunk 5]
```diff
@@ -708,7 +716,7 @@ Container@OBSERVER_WIDGETS:
 						ScrollItem@ECONOMY_PLAYER_TEMPLATE:
 							X: 0
 							Y: 0
-							Width: 640
+							Width: 740
```
Comment (PunkPun) [inline mods/d2k/chrome/ingame-observer.yaml:719]:
```suggestion
							Width: 720
```

you are expanding UI by 80, not 100

**Review by PunkPun — COMMENTED** [review]
The code part looks good, but yaml is a bit of a mess. I've made suggestions

**Comment by PunkPun** [comment]
It might as well be worth switching the places of harvester and carryall, since harvester is the longest word in the UI and it looks better as the last element

**Comment by dragunoff** [comment]
> It might as well be worth switching the places of harvester and carryall, since harvester is the longest word in the UI and it looks better as the last element

IMO it's fine like that and it matches the other mods where oil derricks are in the last column. Besides that @PunkPun's suggestions about the numbers in YAML are correct. This is the result that we're after (right aligned numbers and rows not extending past the last column):
image

**Comment by dragunoff** [comment]
And while we're here should we remove `mods/common/chrome/ingame-observer.yaml`? It's not referenced anywhere :thinking:

**Comment by Mailaender** [comment]
https://cs.github.com/OpenRA/ra2/blob/e655ada6c617e0fd7ec590850c0088c7577ce900/mods/ra2/mod.yaml#L123 and https://cs.github.com/OpenRA/d2/blob/4837e485a9f89308d6a69e2587f11827c018c265/mods/d2/mod.yaml#L198 uses it.

**Comment by PunkPun** [comment]
Ping

**Comment by abcdefg30** [comment]
Closing as superseeded by #20566.

**Pull request closed without merging**